### PR TITLE
ENG-7284 - Refactor Datastore 

### PR DESCRIPTION
**Pull request by kevinsites**
1. Pt 1. Update classes to use dependency injection in order to use same instance and update saveEvent method to add ts, gyro, and accel data
2. Pt 2. Update RN/Android Files to be one file with few util methods
3. Pt 3. Update Datastore to be a class, Update NeuroID to create instance on init, update tests to use NeuroID Instance
     NOTE: This required exposing an extension method called `getTestingDataStoreInstance` which is marked as                  VisibleForTesting. I recommend the re-working of these test to remove the need for this but this PR was large                  enough.
4. Pt.4 - move everything to a single function in the NID class to determine if events should be captured, call the datastore and then do remaining work. Datastore will only capture events and nothing else, added a new method to return if buffer is full

### CODE REVIEW

File: NeuroID/src/main/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt [hunk 1]
```diff
@@ -62,7 +61,14 @@ internal object NIDDataStoreManagerImp : NIDDataStoreManager {
...
 
     @Synchronized
-    override fun saveEvent(event: NIDEventModel) {
+    override fun saveEvent(tempEvent: NIDEventModel) {
+        // add TS, Gyro, and Accel to every event
+        val event = tempEvent.copy(
+            ts = System.currentTimeMillis(),
+            gyro = NIDSensorHelper.getGyroscopeInfo(),
+            accel = NIDSensorHelper.getAccelerometerInfo()
+        )
```
Comment (quacke-nid) [inline NeuroID/src/main/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt:70]: Why not just add these to NIDEventModel construction?

File: NeuroID/src/main/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt [hunk 2]
```diff
@@ -43,15 +42,15 @@ internal object NIDDataStoreManagerImp : NIDDataStoreManager {
...
-    }
-
     // Queue events that are set before sdk is started
     @Synchronized
-    override fun queueEvent(event: NIDEventModel) {
+    override fun queueEvent(tempEvent: NIDEventModel) {
+        val event = tempEvent.copy(
+            ts = System.currentTimeMillis(),
+            gyro = NIDSensorHelper.getGyroscopeInfo(),
+            accel = NIDSensorHelper.getAccelerometerInfo()
```
Comment (quacke-nid) [inline NeuroID/src/main/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt:51]: Why not just add these to NIDEventModel construction?

File: NeuroID/src/main/java/com/neuroid/tracker/callbacks/ActivityCallbacks.kt [hunk 3]
```diff
@@ -3,65 +3,166 @@ package com.neuroid.tracker.callbacks
...
+                NIDEventModel(
+                    type = WINDOW_ORIENTATION_CHANGE,
+                    o = "CHANGED",
+                )
+            )
+            auxOrientation = orientation
+        }
+
+        NIDLog.d(msg="Activity - POST Created - Window Load")
+         dataStore.saveEvent(
```
Comment (quacke-nid) [inline NeuroID/src/main/java/com/neuroid/tracker/callbacks/ActivityCallbacks.kt:102]: indentation

**Review by quacke-nid — APPROVED** [review]

**Comment by kishisaka-nid** [comment]
The NeuroID class is growing pretty large. From the looks of it, The NeuroID class should be a small SDK interface that our customers will use to do stuff. We should consider separating the client interface from our own internal classes that we use. Will think about this. Other than that, pulled, tested, integration tests run locally fine.  🚢

**Review by kishisaka-nid — DISMISSED** [review]

**Comment by kishisaka-nid** [comment]
Looks like we got some unit test failures. Better fix these before merging. > Task :NeuroID:testAndroidLibDebugUnitTest

com.neuroid.tracker.models.NIDEventModelUnitTest > testToJSONSimple FAILED
    org.junit.ComparisonFailure at NIDEventModelUnitTest.kt:12

com.neuroid.tracker.models.NIDEventModelUnitTest > testToJSONAdvanced FAILED
    org.junit.ComparisonFailure at NIDEventModelUnitTest.kt:30

com.neuroid.tracker.service.NIDJobServiceManagerTest > testSendEventsNowSuccess FAILED
    io.mockk.MockKException at NIDJobServiceManagerTest.kt:89

com.neuroid.tracker.service.NIDJobServiceManagerTest > testRestart FAILED
    io.mockk.MockKException at NIDJobServiceManagerTest.kt:81

com.neuroid.tracker.service.NIDJobServiceManagerTest > testSendEventsNowError FAILED
    io.mockk.MockKException at NIDJobServiceManagerTest.kt:115

com.neuroid.tracker.service.NIDJobServiceManagerTest > testStopjob FAILED
    io.mockk.MockKException at NIDJobServiceManagerTest.kt:60

com.neuroid.tracker.service.NIDJobServiceManagerTest > testStartjob FAILED
    io.mockk.MockKException at NIDJobServiceManagerTest.kt:43

com.neuroid.tracker.storage.NIDDataStoreManagerUnitTests > testGetAllEvents FAILED
    io.mockk.MockKException at NIDDataStoreManagerUnitTests.kt:103

com.neuroid.tracker.storage.NIDDataStoreManagerUnitTests > testSaveEvent FAILED
    io.mockk.MockKException at NIDDataStoreManagerUnitTests.kt:83

com.neuroid.tracker.storage.NIDDataStoreManagerUnitTests > testClearEvents FAILED
    io.mockk.MockKException at NIDDataStoreManagerUnitTests.kt:129

**Review by priya-xavier — DISMISSED** [review]
Refactor looks great!

**Comment by priya-xavier** [comment]
Unit tests failing for me too.

**Review by priya-xavier — APPROVED** [review]

**Review by kishisaka-nid — APPROVED** [review]

**Pull request merged**